ui/pages/home/home.component.js

```diff
@@ -392,7 +395,7 @@ export default class Home extends PureComponent {
 
     if (notificationClosing && !prevState.notificationClosing) {
       closeNotificationPopup();
-    } else if (isNotification || hasAllowedPopupRedirectApprovals) {
```

Comment:
note that `isNotification` is a noop / unnecessary guard because it is negated in the `checkStatusAndNavigate()` redirection logic

Reply:
see https://github.com/MetaMask/metamask-extension/blob/6dac3dde6b548c272262407f032c397f4bcc4717/ui/pages/home/home.component.js#L331

Reply:
I guess the fundamental question I have for whoever knows is why the logic was this way before: where any transactions (other than `SNAP_MANAGE_ACCOUNTS_CONFIRMATION_TYPES.showSnapAccountRedirect` confirmations?) that got added to ApprovalController state after initial render would not prompt a redirect to that transaction but rather sit as "unapproved" in the activity list like this:
<img width="362" alt="Screenshot 2024-07-02 at 1 45 59 PM" src="https://github.com/MetaMask/metamask-extension/assets/34557516/8625e37c-1619-4876-902e-5501db3313ca">

This becomes more of a problem in the multi-dapp scenarios (see the before video in the description) which is why we're looking to change it here.

Reply:
I don't see a reason not to. ccing: @matthewwalsh0 in case he has more insight.

Reply:
This might be outside of the scope of the PR, but could it be confusing to the user if we're popping up the confirmation from a previous dapp with no indication that there's another request in the queue?

Reply:
A few clarifications and concerns:

I believed this functionality to be intentional since it prevents usage of the "main UI" from suddenly navigating you away from what you were doing if the dApp suddenly raises a request, supposing you were in the settings for example, or within the send flow. I appreciate it's not instant as we have to redirect to Home first but the popup only shows the first confirmation on load to highlight it, as the badge icon highlights a confirmation is pending.

Where as the notification exists purely to display confirmations, hence why you can very rarely leave a confirmation, and why some of the conditions are specific to notification.

 @jiexi The `isNotification` is only negated for some routing conditions, not all, specifically navigating through the swaps pages which is only supported in the popup.

As an aside, I'm a little confused here as the transaction confirmation includes a navigation component so the user can navigate between multiple queued transactions. I assume this isn't working due to the nature of queued requests and the transactions being on different chains, but could that be a better avenue to support this? Since it would mean the user knows another transaction is queued even while the first transaction is still visible, plus it avoids this blanket logic to allow any new confirmation to re-route the user within the popup.

Reply:
> I believed this functionality to be intentional since it prevents usage of the "main UI" from navigating you away from what you were doing if the dApp raises a request, supposing you were in the settings for example and want to go the send flow via the home page.
> I appreciate it's not instant as we have to redirect to Home first but I understood the popup only shows the first confirmation on load to highlight it, since the badge icon highlights a confirmation is pending.

So part of what has changed with per-dapp selected network that makes the original UX of relying on a badge/activity tab to let you know that there are pending transactions is that now pending transactions could be on a network you're not currently on meaning you could not see this 'unapproved' transaction in the activity tab without figuring out which network the transaction is on and navigating to it.

> I believed this functionality to be intentional since it prevents usage of the "main UI" from navigating you away from what you were doing if the dApp raises a request, supposing you were in the settings for example and want to go the send flow via the home page.

I'm not a UX specialist so will defer here, but this still sounds to me like a case where we'd want to pop the pending transaction and letting the user either confirm or reject before they start a new send flow... What is the case in which you think they'd want to let this transaction sit around unconfirmed until they complete a separate flow?

Reply:
I think I see what you mean regarding the queued requests.

But is the second transaction still in the controller state at the time of the first confirmation, meaning we could still feasibly update selectors to display these transactions from different networks in the standard multiple confirmation navigation? Or is the request itself queued so it's not added to the controllerstate until the first dApp request is processed?

Concerning the use case for not navigating away, I suppose it's similar justification to the full screen condition since my understanding is simply that popup wasn't originally intended to display new confirmations.

But all that said, I'm not necessarily against this, I just wanted to clarify my understanding of the original intent before we change it. Ultimately down to what @bschorchit thinks is best.

I think I've tracked down the original introducion of the notification limitation and it doesn't appear to be justified at the time 😄 

https://github.com/MetaMask/metamask-extension/pull/12065/files#diff-afc7617dc1347d665d4f7a058e9ea8ac686f5ad520874a293f6e27e7c50f9568R193

Reply:
> But is the second transaction still in the controller state at the time of the first confirmation, meaning we could still feasibly update selectors to display these transactions from different networks in the standard multiple confirmation navigation? Or is the request itself queued so it's not added to the controller state until the first dApp request is processed?

Yea if the second transaction is proposed from a different dapp it would not be in ApprovalController state yet

Reply:
Alex walked me through this during a call (thanks!🙏) and we agreed that the user having the popup opened when receiving a dapp request should be an edge case and it didn't seem a big issue to redirect them to the confirmation given the trade-off of not having subsequent confirmations popping up without this change.

Reply:
> This might be outside of the scope of the PR, but could it be confusing to the user if we're popping up the confirmation from a previous dapp with no indication that there's another request in the queue?

Regarding this part, I still have some concern, but will create a separate issue to track and discuss it. It shouldn't block this PR.